### DISCUSSION TITLE
Higher beam size causes Whisper to skip transcribing certain parts. Why?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [openai/whisper#1314] [Q&A]
I have just tested 15 beam_size.

It transcribed almost perfect the parts it didn't fail.

After like 4 minutes there is like 1 minute speech it failed to transcribe.

When beam_size reduced to 5 it transcribed all. Do you know why? How to find a sweet spot?

**Comment by hoonlight** [comment]
You can refer to [this paper](https://arxiv.org/pdf/2204.05424.pdf)

> Many  researchers  observe  that  translation  quality degrades with beam sizes beyond 5 or 10 (Tuet al., 2017; Koehn and Knowles, 2017).  We call this phenomenon the “beam search curse”, which is  listed  as  one  of  the  six  biggest  challenges  for NMT (Koehn and Knowles, 2017).

**Comment by glangford** [comment]
> How to find a sweet spot?

Trial and error until you find the best settings for your language and type of content. I use beam_size 5 and patience 2 with the large model (most of the time)  for the Portuguese language for example, with excellent results.
